ou_dedetai/tui_curses.py

```diff
@@ -2,60 +2,74 @@
 import signal
 import textwrap
 
-from . import config
-from . import msg
-from . import utils
+from ou_dedetai import tui_screen
+from ou_dedetai.app import App
 
 
-def wrap_text(app, text):
+# NOTE to reviewer: does this convay the original meaning?
+# The usages of the function seemed to have expected a list besides text_centered below
+# Which handled the string case. Is it faithful to remove the string case?
+def wrap_text(app: App, text: str) -> list[str]:
+    from ou_dedetai.tui_app import TUI
+    if not isinstance(app, TUI):
+        raise ValueError("curses MUST be used with the TUI")
     # Turn text into wrapped text, line by line, centered
     if "\n" in text:
         lines = text.splitlines()
-        wrapped_lines = [textwrap.fill(line, app.window_width - (config.margin * 2)) for line in lines]
-        lines = '\n'.join(wrapped_lines)
+        wrapped_lines = [textwrap.fill(line, app.window_width - (app.terminal_margin * 2)) for line in lines] #noqa: E501
+        return wrapped_lines
     else:
-        wrapped_text = textwrap.fill(text, app.window_width - (config.margin * 2))
-        lines = wrapped_text.split('\n')
-    return lines
+        wrapped_text = textwrap.fill(text, app.window_width - (app.terminal_margin * 2))
+        return wrapped_text.splitlines()
 
 
-def write_line(app, stdscr, start_y, start_x, text, char_limit, attributes=curses.A_NORMAL):
+def write_line(app: App, stdscr: curses.window, start_y, start_x, text, char_limit, attributes=curses.A_NORMAL): #noqa: E501
+    from ou_dedetai.tui_app import TUI
+    if not isinstance(app, TUI):
+        raise ValueError("curses MUST be used with the TUI")
     try:
         stdscr.addnstr(start_y, start_x, text, char_limit, attributes)
     except curses.error:
-        signal.signal(signal.SIGWINCH, app.signal_resize)
+        # FIXME: what do we want to do to handle this error?
+        # Before we were registering a signal handler
```

Comment:
This exists because of the above: if curses tries to write beyond the screen limits, it will error. The resize is meant to try to correct it by resizing the screen, say, should the screen be moved around the user's desktop, perhaps temporarily shrinking in size, or manually shrinking it down and then back up again.

I'm open to suggestions.

Reply:
pass is probably good enough for now, as we recover from the error